### ISSUE TITLE
GitHub Auth on the ATC

### ISSUE THREAD

**Issue by psellars-hyprnz** [EngineerBetter/concourse-up#61]
Firstly - thanks for such a useful tool, you have helped us get up and running with Concourse across a number of teams will very little pain.

We currently use GitHub Auth with Concourse - which we have been able to set up via the `fly` command line. It appears with 4.0.0 that we need to pass the GitHub Client ID and Secret to the atc now. I assume there are others using other auth abilities with similar issues.

I am assuming too that there are plans afoot to provide access to this via Concourse Up - in the meantime could you suggest a path for us - as not keen to lose our GitHub authentication and it does not appear like the manifest should be changed in this project - as it comes from another EngineerBetter repository (which I don't believe is open sourced - may be wrong on this).

Was also wondering if there are any plans to provide access or capabilities to set other parameters such as for the default resource checking time and default logs to retain?

Am happy/keen to try and contribute changes to provide this - not sure if I have all the skills to do so - but would be happy to try.

**Comment by danyoung** [comment]
Hi @psellars-hyprnz - glad you find the tool useful!  We've created https://github.com/EngineerBetter/concourse-up/issues/62 to track this problem and the solution. You are correct that this affects all auth methods.

A workaround at the moment is to recompile Concourse-Up binary yourself with Github oAuth config in the manifest (mentioned by @adamclark-dev here https://github.com/EngineerBetter/concourse-up/issues/59)

We are working on a solution to the issue and will update as soon as we can. 

Thanks

Dan

**Comment by psellars-hyprnz** [comment]
Hi @danyoung thanks for the response. If I recompile the binary with the manifest changes as mentioned will automatic updates break the deployment?

We are keen to leverage automatic updates - so just checking if the change will mean we need to turn off automatic Concourse Up updates.

Thanks
Pete

**Comment by crsimmons** [comment]
@psellars-hyprnz yes, deploying with a locally built binary then running the auto update will either wipe the changes you've made or break. Once you start modifying the `concourse-up` binary you will need to pause the self-update pipeline and make sure any future updates are done with the modified binary.
